modules/seqexec/web/client/src/main/scala/seqexec/web/client/components/SessionQueueTable.scala

```diff
@@ -720,27 +724,30 @@ object SessionQueueTable extends Columns {
       )
     )
     .configure(Reusability.shouldComponentUpdate)
-    .componentWillReceiveProps { b =>
-      // Reset loading
-      b.modState(_.resetLoading(b.nextProps)) *>
-        // if login state changes recalculate widths
-        b.modStateOption { s =>
-            s.lastSize.map(ls =>
-              (State.userModified.modify { s =>
+    .getDerivedStateFromProps{ (props, state) =>
+      Function.chain(
+        ((s: State) => s.resetLoading(props)) ::
+        List( (s: State) =>
+          s.lastSize.fold(s)( ls =>
+            (
+              State.userModified.modify { um =>
                 // If login state changes discard user modifications
-                if (b.currentProps.loggedIn =!= b.nextProps.loggedIn) {
+                if (props.loggedIn =!= state.prevLoggedIn) {
                   NotModified
-                } else s
-              } >>>
+                } else um
+              } andThen(
                 State.tableState.modify(
-                  _.recalculateWidths(ls, b.nextProps.visibleColumns, b.nextProps.columnWidths)
-                ))(s)
-            )
-          }
-          .when(
-            b.currentProps.obsIds =!= b.nextProps.obsIds || b.currentProps.loggedIn =!= b.nextProps.loggedIn
-          )
-          .void
+                  _.recalculateWidths(ls, props.visibleColumns, props.columnWidths)
+                )
+              )
+            )(s)
+          ),
+          State.prevObsIds.set(props.obsIds),
```

Comment:
Is this why you asked about `zip`for lenses?

Reply:
Yes. I ended up using `Function.chain` with the setters with values.